ironic/drivers/modules/drac/inspect.py

```diff
@@ -44,7 +93,261 @@ class DracRedfishInspect(redfish_inspect.RedfishInspect):
+                    try:
+                        port.create()
+                        LOG.info('Port created with MAC address %(mac)s '
+                                 'for node %(node_uuid)s during inspection',
+                                 {'mac': mac, 'node_uuid': node.uuid})
+                    except exception.MACAlreadyExists:
+                        LOG.warning("Port already exists for MAC address "
+                                    "%(address)s for node %(node)s",
+                                    {'address': mac, 'node': node.uuid})
+            else:
+                LOG.warning("Not attempting to create any port as no NICs "
+                            "were discovered in 'enabled' state for node "
+                            "%(node)s: %(mac_data)s",
+                            {'mac_data': macs, 'node': node.uuid})
+        else:
+            LOG.warning("No NIC information discovered "
+                        "for node %(node)s", {'node': node.uuid})
+
+        return states.MANAGEABLE
+
+    def _get_pxe_dev_nics(self, task, system):
+        """Get a list of pxe device interfaces.
+
+        :param task: a TaskManager instance.
+        :param system: a Redfish System object that represents a node.
+        :returns: Returns list of pxe device interfaces.
+        """
+        pxe_dev_nics = []
+        pxe_params = ["PxeDev1EnDis", "PxeDev2EnDis",
+                      "PxeDev3EnDis", "PxeDev4EnDis"]
+        pxe_nics = ["PxeDev1Interface", "PxeDev2Interface",
+                    "PxeDev3Interface", "PxeDev4Interface"]
+
+        redfish_ip = task.node.driver_info.get('redfish_address')
+        redfish_username = task.node.driver_info.get('redfish_username')
+        redfish_password = task.node.driver_info.get('redfish_password')
+
+        if system.boot.mode == 'uefi':
+            for param, nic in zip(pxe_params, pxe_nics):
+                if system.bios.attributes[param] == 'Enabled':
+                    nic_id = system.bios.attributes[nic]
+                    mac_response = requests.get('%s%s%s' % (
+                        redfish_ip, _MAC_URL, nic_id), auth=(
+                        redfish_username, redfish_password), verify=False)
+                    mac_address = json.loads(mac_response.__dict__.get(
+                        '_content')).get('MACAddress')
+                    pxe_dev_nics.append(mac_address)
+
+        elif system.boot.mode == 'bios':
+            redfish_username = task.node.driver_info.get('redfish_username')
+            redfish_password = task.node.driver_info.get('redfish_password')
+
+            url = "%s%s" % (redfish_ip, _JOB_ID_URL)
+            job_response = requests.post(url, data=json.dumps(
+                _PAYLOAD), headers=_HEADERS, verify=False, auth=(
+                redfish_username, redfish_password))
+
+            if job_response.status_code != _JOB_RESPONSE_CODE:
+                LOG.debug("FAIL, status code not 202, code is: %(code), "
```

Comment:
I would think we would want to raise an exception here instead of just returning an empty list for pxe_dev_nics

Reply:
maybe a HardwareInspectionFailure exception?

Reply:
Done